lib/src/errors.rs

```diff
@@ -1,19 +1,60 @@
+use std::error;
+use std::fmt;
+
 use telegram_bot_raw;
 
-error_chain! {
-    foreign_links {
-        Url(::hyper::http::uri::InvalidUri);
-        Hyper(::hyper::Error);
-        Http(::hyper::http::Error);
-        InvalidHeaderValue(::hyper::http::header::InvalidHeaderValue);
-        Io(::std::io::Error);
+#[derive(Debug)]
+pub struct Error(ErrorKind);
+
+#[derive(Debug)]
+pub(crate) enum ErrorKind {
```

Comment:
I think this could benefit from being public for all

Reply:
Any plans to open this to public? I currently match on `ErrorKind::Raw` to determine if it's a Telegram API error, which is no longer possible after this PR.

Reply:
We can add something like
```rust
// In telegram-bot-raw
pub struct TelegramError {
    description: String,
    parameters: Option<ResponseParameters>,
}

// In telegram-bot
impl Error {
    fn as_telegram_error(&self) -> Option<&TelegramError> {...}
}
```

What do you think?

Reply:
That should work for my use case. Is there a reason why we are keeping the type opaque? As a library, it might be useful to allow developers to inspect the error for error handling.